mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -696,3 +702,64 @@ def update_artifacts_location(self, run_id, new_artifacts_location):
         with self.ManagedSessionMaker() as session:
             run = session.query(SqlRun).filter_by(run_uuid=run_id).first()
             run.artifact_uri = new_artifacts_location
+
+
+def _get_attributes_filtering_clauses(parsed):
+    clauses = []
+    for sql_statement in parsed:
+        key_type = sql_statement.get('type')
+        key_name = sql_statement.get('key')
+        value = sql_statement.get('value')
+        comparator = sql_statement.get('comparator')
+        if SearchUtils.is_attribute(key_type, comparator):
+            op = SearchUtils.filter_ops.get(comparator)
+            if op:
+                clauses.append(op(getattr(SqlRun, key_name), value))
+    return clauses
+
+
+def _to_sqlalchemy_filtering_statement(sql_statement, session):
+    key_type = sql_statement.get('type')
+    key_name = sql_statement.get('key')
+    value = sql_statement.get('value')
+    comparator = sql_statement.get('comparator')
+
+    if SearchUtils.is_metric(key_type, comparator):
+        entity = SqlLatestMetric
+        entity_key = SqlLatestMetric.key
+        entity_value = SqlLatestMetric.value
+        value = float(value)
+    elif SearchUtils.is_param(key_type, comparator):
+        entity = SqlParam
+        entity_key = SqlParam.key
+        entity_value = SqlParam.value
+    elif SearchUtils.is_tag(key_type, comparator):
+        entity = SqlTag
+        entity_key = SqlTag.key
+        entity_value = SqlTag.value
+    elif SearchUtils.is_attribute(key_type, comparator):
+        return None
+    else:
+        raise MlflowException("Invalid search expression type '%s'" % key_type,
+                              error_code=INVALID_PARAMETER_VALUE)
+    op = SearchUtils.filter_ops.get(comparator)
```

Comment:
Is there a valid case in which a supplied `comparator` is not contained in the set of supported `SearchUtils.filter_ops`? It seems like this case would imply that the `comparator` is invalid, in which case we should probably throw an exception.